Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -2272,6 +2315,68 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
             }
         }
 
+        if (rightConst && rightConst.isAnyInt()) {
```

Comment:
Just out of curiosity, can we use same approach for Float TypedArray family, `ArrayWithInt32` and `ArrayWithDouble` ?

Reply:
Yes, we can extend this. But for now, starting with simpler ones first.